randutil/crypto.go

```diff
@@ -49,13 +49,16 @@ func CryptoToken(nbytes int) (string, error) {
 	return base64.RawURLEncoding.EncodeToString(b), nil
 }
 
+// Allow mocking of the path for testing purposes.
+var KernelUUIDPath = "/proc/sys/kernel/random/uuid"
+
 // RandomKernelUUID will return a UUID from the kernel's procfs API at
 // /proc/sys/kernel/random/uuid. Only to be used in very specific uses, most
 // random code should use CryptoToken(Bytes) instead.
-func RandomKernelUUID() string {
-	b, err := ioutil.ReadFile("/proc/sys/kernel/random/uuid")
+func RandomKernelUUID() (string, error) {
+	b, err := ioutil.ReadFile(KernelUUIDPath)
 	if err != nil {
-		panic("cannot read kernel generated uuid")
+		return "", err
```

Comment:
I wonder if this should be prefixed with some context like: `cannot generate random kernel uuid: %v" to give the user a bit more context (but maybe it's fine as is)